filecoin-proofs/tests/api.rs

```diff
@@ -52,132 +52,164 @@ use filecoin_proofs::{
 // same porep_ids).
 const ARBITRARY_POREP_ID_V1_0_0: [u8; 32] = [127; 32];
 const ARBITRARY_POREP_ID_V1_1_0: [u8; 32] = [128; 32];
+const ARBITRARY_POREP_ID_V1_2_0: [u8; 32] = [129; 32];
 
 const TEST_SEED: [u8; 16] = [
     0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc, 0xe5,
 ];
 
 #[test]
 #[ignore]
-fn test_seal_lifecycle_2kib_porep_id_v1_base_8() -> Result<()> {
-    let porep_id_v1: u64 = 0; // This is a RegisteredSealProof value
+fn test_seal_lifecycle_2kib_base_8() -> Result<()> {
+    // The first value is RegisteredSealProof value
+    // The second value is the ApiVersion to use
+    let test_inputs = vec![
+        (0, ApiVersion::V1_0_0),
+        (5, ApiVersion::V1_1_0),
+        (5, ApiVersion::V1_2_0),
+    ];
 
-    let mut porep_id = [0u8; 32];
-    porep_id[..8].copy_from_slice(&porep_id_v1.to_le_bytes());
-    assert!(is_legacy_porep_id(porep_id));
-    seal_lifecycle::<SectorShape2KiB>(SECTOR_SIZE_2_KIB, &porep_id, ApiVersion::V1_0_0)
-}
+    for input in test_inputs {
+        let porep_id_num = input.0 as u64;
+        let api_version = input.1;
 
-#[test]
-#[ignore]
-fn test_seal_lifecycle_2kib_porep_id_v1_1_base_8() -> Result<()> {
-    let porep_id_v1_1: u64 = 5; // This is a RegisteredSealProof value
+        let mut porep_id = [0u8; 32];
+        porep_id[..8].copy_from_slice(&porep_id_num.to_le_bytes());
 
-    let mut porep_id = [0u8; 32];
-    porep_id[..8].copy_from_slice(&porep_id_v1_1.to_le_bytes());
-    assert!(!is_legacy_porep_id(porep_id));
-    seal_lifecycle::<SectorShape2KiB>(SECTOR_SIZE_2_KIB, &porep_id, ApiVersion::V1_1_0)
-}
+        assert!(match api_version {
+            ApiVersion::V1_0_0 => is_legacy_porep_id(porep_id),
+            ApiVersion::V1_1_0 | ApiVersion::V1_2_0 => !is_legacy_porep_id(porep_id),
+        });
```

Comment:
I propose refactoring this into a function like:

```rust
fn to_porepid_verified(registered_seal_proof: u64, api_version: ApiVersion) -> [u8; 32] {
    let mut porep_id = [0u8; 32];
    porep_id[..8].copy_from_slice(&registered_seal_proof.to_le_bytes());
    assert!(match api_version {
        ApiVersion::V1_0_0 => is_legacy_porep_id(porep_id),
        ApiVersion::V1_1_0 | ApiVersion::V1_2_0 => !is_legacy_porep_id(porep_id),
    }
    porep_id
}
```

Which then can be used (with the comments above applied) within this loop as:

```rust
let porep_id = to_porepid_verified(registered_seal_proof, api_version);
```